### ISSUE TITLE
Vis crystal bugs

### ISSUE THREAD

**Issue by EnderiumSmith** [Azanor/thaumcraft-beta#802]
1) The default spawn rate is way to high. It is so high that it breaks mob spawning in twilight forest dungeons by lighting over half of it. Its not uncommon to see 3 different kinds in the same place.

2) Crystals try to spawn on nonsolid blocks like cobble walls and pop off.

3) The opposite is also true: the crystals count as solid blocks so fences connect to them and if a mob walks on one it gets stuck spinning in circles.

**Comment by Geethebluesky** [comment]
A dimension blacklist or per-dimension config would be great to solve point 1.

I've also seen 3 happen.

**Comment by EnderiumSmith** [comment]
but they should spawn in the twilight forest. maybe make them only spawn on stone so they dont spawn in the dungeons.

**Comment by redgoblin88** [comment]
the crystals are meant to be like "staligtites" or "staligmites" in which it is additional excessive vis in the area clumping up and crystalising on a surface. it makes sense that it can form on any block (even leaves) as its like a sap or crystalline substance which is designed to clump together and accumulate over time absorbing vis and growing bigger

**Comment by Geethebluesky** [comment]
Ah so you just want the crystals to be in tune with the dimension itself, yeah. That makes sense too.

I guess I prefer the idea where the TF is a different kind of world that doesn't have all the properties of the Overworld but that might be too much to implement :)

**Comment by redgoblin88** [comment]
well it would be handy if there was a config setting that you could increase or decrease the chance for them spawning, or increase/decrease the vis in a dimention

**Comment by Geethebluesky** [comment]
Yeah, someone on Reddit was wondering if oreDensity changes the crystal spawn rate but nobody answered.

**Comment by EnderiumSmith** [comment]
@redgoblin88 while i agree that you should be able to place them on anything, they should not spawn on dungeon blocks at worldgen. maybe a blacklist for blacks they cant spawn on would work better

**Comment by Azanor** [comment]
I can make them emit light level 1 so they will emit some light, but not enough to prevent spawns.

**Comment by EnderiumSmith** [comment]
making them render full bright would be better. i assume its already a custom model since it has complex angles so it should be as simple as changing the inputs for the vertex things.

As for light level the max amount seems to be 5 so a max light of 5 would work. can mobs spawn IN them like they can with snow?
